### PR TITLE
Automate start of QEMU

### PR DESCRIPTION
**Pull request by svlad-90**
doma, domu: Add dependency to qemu-restart-monitor
In virtio build domu and doma depend on additional service Yocto package
- qemu-restart-monitor.

This patch covers that additional dependency.

This change is temporary. It should be modified, reduced or even totally
removed when Xen will add internal support to start QEMU as provider of
virtio backends for the guest domain.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

domu: Remove XT_DOMU_DTB_NAME for virtio build
In case of virtio build the domu domain remains generic. It does not
require any additional dtb overlay in order to work properly. The dtb
generated by Xen is fully sufficient.

This patch is removing usage of the XT_DOMU_DTB_NAME variable, which was
causing Yocto to search for the dtb 'salvator-generic-domu.dtb' file,
which is provided by the Renesas BSP, and which does not exist in terms
of virtio build. That was causing a build failure of the DomU Yocto
build.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

yaml: Normalize MACHINE names for the virtio build
In virtio we need to use the following machine names for the build:

- Dom0 - generic-armv8-xt
- DomU - virtio-armv8-xt
- DomD - %{MACHINE}

DomD remains the only Xen domain, which is hardware-specific and is
using the board-specific BSP meta-layers.

This patch ensures that the correct machine names are used by each
domain.

Moreover, it simplifies management of the machine names for each domain,
as the yaml declarations are refactored to re-use the values of the
following 3 variables everywhere:

- DOM0_MACHINE
- DOMD_MACHINE
- DOMU_MACHINE

Thus, if needed in the future, it would be easy to change each machine's
name without the need to directly copy-paste literals all over the
document.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

README: Update virtio documentation
We've switched from the ENABLE_VIRTIO moulin parameter to a separate
yaml file. It is time to update the product's documentation to reflect that
and to provide more insights regarding what is the difference between
virtio and non-virtio builds.

This patch updates README.md to provide the above-mentioned information.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

dom-0: update xen config for dom-a
Update dom-a xen configuration for 'virtio' build configuration.

For virtio functionality, we need to use different xen configuration
in comparison with the non-virtio builds.

This patch introduces dynamic filling of the dom-a's xen configuration
depending on 'enable_virtio' distro feature.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

doma: Introduced dts for virtio build
For virtio builds we do not need gsx and ion related information in the
dts for DomA.

This patch introduces the virtio-specific DomA dts -
salvator-generic-doma-virtio.dts

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

yaml: Add enable_virtio distro feature to domu
In non-virtio builds we inject additional weston configuration to domu
using the meta-xt-rcar->meta-rcar-gen3-adas meta-layer.

For virtio builds we do not use any of the renesas specific meta-layers
in domu. But we still need that additional weston configuration. Thus,
we add it in the meta-xt-common->meta-xt-domu. And that is done only
when 'enable_virtio' distro feature and OVERRIDES are turned on.

This patch activates "enable_virtio" distro feature and OVERRIDES for
the virtio builds of domu Xen domain.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

doma & domu: Add support of NFS for virtio builds
We had the following hardcoded parts in our product:
- The usage of the '/dev/mmcblk0p3' literal in QEMU command for virtio
builds. It can have another name
- The usage of the 'virtio-blk-pci' QEMU device even in case of the NFS boot
for virtio builds. It is not needed when boot is done from the NFS
- An unconditional usage of the 'xvda1' search literal for NFS boot in
domu-set-root. In virtio we have a 'vda' root device for domu, so should
search for another literal

This patch is intended to consider and resolve all the above findings.

This change is temporary. It should be modified, reduced or even totally
removed when Xen will add internal support to start QEMU as provider of
virtio backends for the guest domain.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

### CODE REVIEW

**Comment by svlad-90** [comment]
The changes were verified in the following way:

|Build type|Comment|
|Non-virtio DomU|Boot is successful. Graphics works.|
|Non-virtio DomA|Boot is successful. Graphics works.|
|Virtio DomU|All expected systemd services are running. The system does not boot due to the lack of the proper Xen-configuration.|
|Virtio DomA|All expected systemd services are running. The system does not boot due to the lack of the proper Xen-configuration.|

File: meta-xt-prod-devel-rcar-driver-domain/recipes-graphics/recipes-graphucs/wayland/weston-init.bbappend [hunk 1]
```diff
@@ -0,0 +1,10 @@
+do_install:append() {
+    # Turn off usage of the LVDS connector in case of virtio
+    if ${@bb.utils.contains('DISTRO_FEATURES', 'enable_virtio', 'true', 'false', d)}; then
+        sed -e '$a\\' \
```
Comment (lorc) [inline meta-xt-prod-devel-rcar-driver-domain/recipes-graphics/recipes-graphucs/wayland/weston-init.bbappend:4]: Is this a fancy way to write
```
echo "[output] 
name=LVDS-1
mode=off" >>  ${D}/${sysconfdir}/xdg/weston/weston.ini
```
? :)

**Review by lorc — COMMENTED** [review]
There is no big comments for this PR, but I expect you to rework "doma, domu: Add declarations for automated start of QEMU" taking into account my comments for meta-xt-common PR. 

Also, this layer is the correct place to store domain configuration. Hint: take a look at domu-set-root script.

**Comment by svlad-90** [comment]
Hi @lorc, I've added an additional commit, which updates the virtio part of the documentation.

File: meta-xt-prod-devel-rcar-driver-domain/recipes-graphics/recipes-graphucs/wayland/weston-init.bbappend (hunk 1 above)
Comment (lorc) [reply meta-xt-prod-devel-rcar-driver-domain/recipes-graphics/recipes-graphucs/wayland/weston-init.bbappend:4]: Can I get some answer, please?

Comment (svlad-90) [reply meta-xt-prod-devel-rcar-driver-domain/recipes-graphics/recipes-graphucs/wayland/weston-init.bbappend:4]: Hi @lorc, this, as all other minor findings here and in meta-xt-common, are accepted and will be fixed.

I'm trying to resolve your big comment which asks me to change "many huge things". Trying to find motivation for doing that first. ))

**Comment by svlad-90** [comment]
Hi @lorc, I've pushed new 2 commits:
https://github.com/xen-troops/meta-xt-prod-devel-rcar/commit/6e68a5cb0c2545a84d58cfad5c54be32807f3565
https://github.com/xen-troops/meta-xt-prod-devel-rcar/commit/946d03305b08d1e54f6972528aa3147c5eaf4c5b

This is totally the same thing, which DTerletskyi has done for DomU here:
https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/100
But only for DomA. And already in a format that was approved by you for DomU. So, hopefully, there won't be additional comments from your side regarding these 2 commits. ))

It should finally unblock the testing of virito based on the xen-troops master branches.

I've decided to add it here, as I must change a lot to react to your findings, so I will need to do additional testing. So both DomA and DomU should be running in order to validate the changed implementation.

=> This message is just to say that I haven't yet reacted to your findings. I'm doing some additional work to unblock that.

**Comment by svlad-90** [comment]
Hi @lorc, this PR is ready for another iteration of the review. Please, have a look.

**Comment by svlad-90** [comment]
Another PR will be created instead of this one, as this is already overloaded with comments.

**Pull request closed without merging**